File: pjecalc-fonte/base/br/jus/trt8/pjecalc/base/comum/HelperIterate.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.base.comum;

import br.jus.trt8.pjecalc.base.comum.Utils;
import java.util.Collection;

public class HelperIterate<E> {
    private Collection<E> collection;

    private HelperIterate(Collection<E> collection) {
        this.collection = collection;
    }

    public static <E> HelperIterate<E> iterate(Collection<E> collection) {
        return new HelperIterate<E>(collection);
    }

    public boolean where(Where<E> where) {
        if (Utils.naoNulo(this.collection)) {
            for (E item : this.collection) {
                if (!where.evaluate(item)) continue;
                return true;
            }
        }
        return false;
    }

    public static interface Where<E> {
        public boolean evaluate(E var1);
    }
}

File: src/lib/pjecalc/core/base/comum/helper-iterate.ts
/**
 * PJe-Calc v2.15.1 — HelperIterate
 * Porte 1:1 de: br.jus.trt8.pjecalc.base.comum.HelperIterate
 *
 * Wrapper que encapsula `collection.some(predicate)`. Em Java serve como
 * açúcar sintático; em TS é redundante com `Array.prototype.some`, mas
 * mantemos para preservar simetria quando o código cliente for portado.
 *
 * Uso fluente:
 *   HelperIterate.iterate(items).where((i) => i.valor > 0);
 *
 * Ref Java: pjecalc-fonte/base/br/jus/trt8/pjecalc/base/comum/HelperIterate.java
 */
import { naoNulo } from './utils';

/** Predicate para uso com HelperIterate.where. */
export type Where<E> = (item: E) => boolean;

export class HelperIterate<E> {
  private readonly collection: Iterable<E> | null | undefined;

  private constructor(collection: Iterable<E> | null | undefined) {
    this.collection = collection;
  }

  /** HelperIterate.iterate (linha 16) — entry point fluente. */
  static iterate<E>(collection: Iterable<E> | null | undefined): HelperIterate<E> {
    return new HelperIterate<E>(collection);
  }

  /**
   * HelperIterate.where (linha 20).
   * Retorna TRUE se ao menos um elemento satisfaz `where`.
   * Coleção null → FALSE (sem iteração).
   */
  where(where: Where<E>): boolean {
    if (!naoNulo(this.collection)) return false;
    for (const item of this.collection as Iterable<E>) {
      if (where(item)) return true;
    }
    return false;
  }
}
